scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -152,29 +152,39 @@ abstract class ScanResultsStorage {
                 "Password for PostgreSQL storage is missing."
             }
 
-            val properties = Properties()
-            properties["user"] = config.username
-            properties["password"] = config.password
-            properties["ApplicationName"] = "$ORT_FULL_NAME - $TOOL_NAME"
-
-            // Configure SSL, see: https://jdbc.postgresql.org/documentation/head/connect.html
-            // Note that the "ssl" property is only a fallback in case "sslmode" is not used. Since we always set
-            // "sslmode", "ssl" is not required.
-            properties["sslmode"] = config.sslmode
-            config.sslcert?.let { properties["sslcert"] = it }
-            config.sslkey?.let { properties["sslkey"] = it }
-            config.sslrootcert?.let { properties["sslrootcert"] = it }
-
-            val connection = DriverManager.getConnection(config.url, properties)
+            val dataSourceConfig = HikariConfig().apply {
+                jdbcUrl = config.url
+                username = config.username
+                password = config.password
+                // use a value slightly bigger than the number of threads accessing the storage
+                maximumPoolSize = 8
+                addDataSourceProperty("ApplicationName", "$ORT_FULL_NAME - $TOOL_NAME")
+
+                // Configure SSL, see: https://jdbc.postgresql.org/documentation/head/connect.html
+                // Note that the "ssl" property is only a fallback in case "sslmode" is not used. Since we always set
+                // "sslmode", "ssl" is not required.
+                addDataSourceProperty("sslmode", config.sslmode)
+                addNullableProperty("sslcert", config.sslcert)
+                addNullableProperty("sslkey", config.sslkey)
+                addNullableProperty("sslrootcert", config.sslrootcert)
+            }
 
-            return PostgresStorage(connection, config.schema).also { it.setupDatabase() }
+            return PostgresStorage(HikariDataSource(dataSourceConfig), config.schema).also { it.setupDatabase() }
         }
 
         /**
          * Create a [ClearlyDefinedStorage] based on the [config] passed in.
          */
         private fun createClearlyDefinedStorage(config: ClearlyDefinedStorageConfiguration): ScanResultsStorage =
             ClearlyDefinedStorage(config)
+
+        /**
+         * Add a property with the given [key] and [value] to the [HikariConfig]. If the [value] is *null*, this
+         * function has no effect.
```

Comment:
Please explain why this helper function is necessary, i.e. what happens if you do set a value to `null` otherwise?

Reply:
Added a sentence about *null* values.